src/Microsoft.AspNetCore.Mvc.Core/Internal/DefaultAssemblyPartDiscoveryProvider.cs

```diff
@@ -101,23 +100,28 @@ private Dependency CreateDependency(RuntimeLibrary library, ISet<string> referen
 
             private DependencyClassification ComputeClassification(string dependency)
             {
-                Debug.Assert(_dependencies.ContainsKey(dependency));
+                if (!_runtimeDependencies.ContainsKey(dependency))
```

Comment:
So the scenario that triggered it was unusual, but here's the line in the deps file the user had: https://github.com/Gonkers/Github-Issues/blob/master/Files/ConsoleApp1/WebApplication1/bin/Debug/PublishOutput/WebApplication1.deps.json#L2646

In this case, `Library.Dependencies` for Kestrel would list `Libuv`, but `Libuv` does not have a runtime component. The assert fails and throws a KeyNotFoundException